app/services/VatReturnService.scala

```diff
@@ -61,33 +63,38 @@ class VatReturnService @Inject()(
       lastUpdated = Instant.now(clock)
     )
 
-    val insertPayload = repository.insert(vatReturn)
-
-    sendToCoreIfEnabled(vatReturn, emptyCorrectionPayload, insertPayload)
+    sendToCoreIfEnabled(vatReturn, emptyCorrectionPayload, repository.insert(vatReturn))
   }
 
-  private def sendToCoreIfEnabled[A](vatReturn: VatReturn, correctionPayload: CorrectionPayload, block: => Future[A])(implicit hc: HeaderCarrier): Future[A] = {
+  private def sendToCoreIfEnabled[A](vatReturn: VatReturn, correctionPayload: CorrectionPayload, block: => Future[A])(implicit hc: HeaderCarrier): Future[Either[CoreErrorResponse, A]] = {
     if (appConfig.coreVatReturnsEnabled) {
 
       for {
         coreVatReturn <- coreVatReturnService.toCore(vatReturn, correctionPayload)
         submissionResult <- coreVatReturnConnector.submit(coreVatReturn).flatMap {
           case Right(_) =>
             logger.info("Successful submission of vat return to core")
-            block
-          case Left(coreErrorResponse) =>
+            block.map(
+              payload => Right(payload)
+            )
+          case Left(coreErrorResponse) if(coreErrorResponse.error == REGISTRATION_NOT_FOUND) =>
+            logger.error(s"Error occurred while submitting to core $coreErrorResponse", coreErrorResponse.asException)
+            Future.successful(Left(coreErrorResponse))
+          case Left(coreErrorResponse) if(coreErrorResponse.error != REGISTRATION_NOT_FOUND) =>
```

Comment:
I think the if statement is redundant here

Reply:
removed